### DISCUSSION TITLE
Lesson 1 Declaration error

### DISCUSSION THREAD

**Discussion by JacobG321** [smartcontractkit/full-blockchain-solidity-course-py#520] [Q&A] [answered]
receiving a declaration error for this line of code, undeclared identifier.
`
 People public person = People({favoriteNumber: 2, name: Patrick});`

full thing below

```Solidity

pragma solidity ^0.6.0;

contract SimpleStorage {

    //this will get initialized to 0!
    uint256 favoriteNumber;

    struct People {
        uint256 favoriteNumber;
        string name;
    }

    People public person = People({favoriteNumber: 2, name: Patrick});

    function store(uint256 _favoriteNumber) public {
        favoriteNumber = _favoriteNumber;
    }
    // view, pure
    function retrieve() public view returns(uint256) {
        return favoriteNumber;
    }

}
```

**Comment by cromewar** [comment, marked as answer]
Hello @TwoDogsOneBigOneSmall 
Solidity does not support default parameters, so setting up `{favoriteNumber: 2, name: Patrick}` is not valid. You should set those values at execution, not at declaration. However, there is a workaround if you want to set default values. you can refer to [this](https://stackoverflow.com/questions/52113202/how-to-set-default-parameters-to-functions-in-solidity) thread.

Best regards, Cromewar.

**Reply by JacobG321** [reply]
thank you so much!

**Reply by cromewar** [reply]
Fell free to make any other questions, I'll close this thread now.

Best regards, Cromewar.
